File: articles/trade-offs-values-and-competing-objectives/python/run_all_tradeoff_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Trade-Offs, Values, and Competing Objectives."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "weighted_objective_model.py",
    "dominance_analysis.py",
    "rank_stability_analysis.py",
    "scenario_regret_analysis.py",
    "stakeholder_value_profiles.py",
    "tradeoff_review_queue.py",
    "decision_record_exporter.py",
    "tradeoffs_values_competing_objectives_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python trade-off workflows completed.")


if __name__ == "__main__":
    main()

File: articles/trade-offs-values-and-competing-objectives/r/run_all_tradeoff_workflows.R
# run_all_tradeoff_workflows.R
# Run all R workflows for Trade-Offs, Values, and Competing Objectives.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "tradeoffs_values_competing_objectives_workflow.R",
  "weighted_objective_tables.R",
  "dominance_review_tables.R",
  "rank_stability_tables.R",
  "scenario_regret_tables.R",
  "tradeoff_review_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R trade-off workflows completed.")
